pkg/types/server_types.go

```diff
@@ -156,10 +157,16 @@ func (r *APIRequest) Option(key string) string {
 }
 
 func (r *APIRequest) WriteResponse(code int, obj APIObject) {
+	for _, warning := range obj.Warnings {
+		r.Response.Header().Add("Warning", fmt.Sprintf("%d %s %s", warning.Code, warning.Agent, warning.Text))
```

Comment:
Nit: looks like the K8s format uses a dash after the code. For example:
`Warning: 299 - "policy/v1beta1 PodSecurityPolicy is deprecated in v1.21+, unavailable in v1.25+"`
But I am not sure how the agent's string comes into play here.

Reply:
See https://github.com/rancher/apiserver/pull/19#discussion_r1055731462  the agent may itself just be the string "-"